### PR TITLE
498 Ability to delete accounts

### PR DESCRIPTION
**Pull request by NeoLou**
This PR is about adding a feature to delete user accounts in the page "frosh-info-table":

- [x] Add new scope for deleting accounts to frontend
- [x] Add new account delete route (with user id)
- [x] Add delete column to frosh data table page with X icon and functionality to frontend (and if correct auth scopes)
- [x] Add delete confirmation popup on frontend before proceeding

To test this you will need to have the scopes to view and delete info in the frosh-info-table (`"accounts:delete", "accounts:edit", "accounts:read", "admin:all", "froshData:unRegisteredUsers", "froshGroupData:all"`). I don't actually know if you need all these scopes but I had them when I tested it myself.

Steps to test:
- Add the following auth-scopes to your user: `"accounts:delete", "accounts:edit", "accounts:read", "admin:all", "froshData:unRegisteredUsers", "froshGroupData:all"`
- Go to `http://localhost:3000/frosh-info-table`
- If you can already see some users, try to delete one of them by pressing the 'X' button in the "Delete Accounts" column
- If you don't see any users or only see one, create more users first before trying to delete
- Try to test deleting both leedur, frosh, and unregistered user accounts
- Check if the user is deleted in the database :)

### CODE REVIEW

**Review by natapokie — APPROVED** [review]
looks great neo!! the functionality works as expected with frosh, leedurs and non-registered frosh!

i don't know if this is a bug, but you can actually delete the account that you're logged into, I think it's fine though haha

File: server/src/services/UserServices.js [hunk 1]
```diff
@@ -297,6 +297,23 @@ const UserServices = {
...
+   * Hard deletes a user by id.
+   * @param {ObjectId} id - id of the user to be deleted
+   * @async
+   * @return {Promise<Object>} - the user which was deleted
+   */
+  async deleteUser(id) {
+    return new Promise((resolve, reject) => {
+      UserModel.findOneAndDelete({ _id: id }, (err, deletedUser) => {
+        if (err || !deletedUser) {
+          reject('UNABLE_TO_DELETE_Timeline');
```
Comment (Cali0707) [inline server/src/services/UserServices.js:310]:
```suggestion
          reject('UNABLE_TO_DELETE_USER');
```

**Review by Cali0707 — CHANGES_REQUESTED** [review]
@NeoLou one small change and then this is ready to merge

**Comment by NeoLou** [comment]
@jameskokoska any other changes you think I should make?

**Comment by NeoLou** [comment]
@Cali0707 made the change!

**Comment by jameskokoska** [comment]
@natapokie could you do one final review and merge

**Review by natapokie — APPROVED** [review]

**Comment by natapokie** [comment]
> @natapokie could you do one final review and merge

I think @Cali0707 needs to approve the requested changes first (?)

**Comment by jameskokoska** [comment]
No problem, if it looks good to you then go ahead and merge it

**Comment by natapokie** [comment]
> No problem, if it looks good to you then go ahead and merge it

merging is blocked 😅

**Pull request merged**